hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/StreamReadMonitoringFunction.java

```diff
@@ -178,6 +200,14 @@ public void initializeState(FunctionInitializationContext context) throws Except
               getClass().getSimpleName(), issuedInstant, issuedOffset, conf.get(FlinkOptions.TABLE_NAME), path);
         }
       }
+      Iterator<SplitState> inputSplitsIterable = inputSplitsState.get().iterator();
+      if (inputSplitsIterable.hasNext()) {
+        SplitState splitState = inputSplitsIterable.next();
+        for (MergeOnReadInputSplit split : splitState.getRemainingSplits()) {
```

Comment:
can use `remainingSplits.addAll(splitState.getRemainingSplits());` which internally utilizes the efficient System.arraycopy.